### PR TITLE
disk_format: New case qemu_img_convert_with_copy_offloading

### PR DESCRIPTION
**Pull request by hellohellenmao**
Add new case for 'qemu-img convert' with copy_offloading

Signed-off-by: Tingting Mao <timao@redhat.com>

### CODE REVIEW

File: qemu/tests/cfg/qemu_img_convert_with_copy_offloading.cfg [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    virt_test_type = qemu
+    type = qemu_img_convert_with_copy_offloading
+    start_vm = no
+    required_qemu = [2.10.0,)
+    remove_image = yes
+    src_image = "source"
+    tgt_image = "target"
+    image_name_source = "images/source"
+    image_name_target = "images/target"
+    image_target_format = "qcow2 raw"
```
Comment (lolyu) [inline qemu/tests/cfg/qemu_img_convert_with_copy_offloading.cfg]: I believe it is better to split `image_target_format` into variants.

**Comment by hellohellenmao** [comment]
@YongxueHong Help to take a review,please.

**Comment by hellohellenmao** [comment]
For qcow2:
(1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading: PASS (4.75 s)

For raw:
(1/1) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading: PASS (4.44 s)

**Comment by hellohellenmao** [comment]
@lolyu @YongxueHong   Help to review again, please.

**Comment by hellohellenmao** [comment]
For raw - Pass.
 (1/2) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading.img_fmt_tgt_raw: PASS (6.66 s)
 (2/2) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading.img_fmt_tgt_qcow2: PASS (6.53 s)

For qcow2 - Pass.
 (1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading.img_fmt_tgt_raw: PASS (6.80 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading.img_fmt_tgt_qcow2: PASS (7.15 s)

**Comment by hellohellenmao** [comment]
@lolyu @yhong HELP TO TAKE A REVIEW, PLS.

File: qemu/tests/qemu_img_convert_with_copy_offloading.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        QemuIOSystem(test, params, img.image_filename).cmd_output(cmd, 120)
+
+    def _convert_with_copy_offloading_and_verify(src, tgt):
+        """Verify whether copy_offloading works."""
+        logging.info("Verify whether copy_offloading works for commit.")
+        cmd = ("strace -e trace=copy_file_range -f qemu-img convert -C -f "
+               "%s %s -O %s %s " % (src.image_format, src.image_filename,
+                                    tgt.image_format, tgt.image_filename))
+        logging.info("Running '%s'." % cmd)
+        sts, text = process.getstatusoutput(cmd)
```
Comment (lolyu) [inline qemu/tests/qemu_img_convert_with_copy_offloading.py]: better to log the output.

File: qemu/tests/cfg/qemu_img_convert_with_copy_offloading.cfg [hunk 3]
```diff
@@ -0,0 +1,17 @@
+- qemu_img_convert_with_copy_offloading:
+    virt_test_type = qemu
+    type = qemu_img_convert_with_copy_offloading
+    start_vm = no
+    required_qemu = [2.10.0,)
+    remove_image = yes
```
Comment (lolyu) [inline qemu/tests/cfg/qemu_img_convert_with_copy_offloading.cfg]: image `source` and `target` are not deleted.

**Comment by hellohellenmao** [comment]
(1/2) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading.img_fmt_tgt_raw: PASS (6.40 s)
 (2/2) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_copy_offloading.img_fmt_tgt_qcow2: PASS (6.61 s)

**Pull request merged**